### ISSUE TITLE
authentication: remote functionality

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#3444]
Follow on to #3359 
Best attempted after #3445, #3449

Quick fix to re-instate remote functionality:

* Ensure only relevent keys are coppied to remote hosts.
  * Server public
  * Client private
* Perform any modifications to the code required to allow auth to function with just these keys.
* As a temporary measure (until #3443) we can allow private keys to travel.

This work is complete when the remote tests pass:

```console
$ git grep --name-only set_test_remote
```

**Comment by oliver-sanders** [comment]
Note: We consider this to be high priority so we can reinstate remote functionality before breakages creep in. This is also a blocker to `rose suite-run` migration which is due to begin shortly.

**Comment by dwsutherland** [comment]
Please put this on hold until #3389 is merged, because the network code has been refactored (due to the new PUB/SUB pattern using the same security/logic).

**Comment by hjoliver** [comment]
(#3389 is merged)
